synapse/handlers/auth.py

```diff
@@ -1226,6 +1253,39 @@ def _complete_sso_login(
         )
         respond_with_html(request, 200, html)
 
+    async def _sso_login_callback(self, login_result: JsonDict) -> None:
+        """
+        A login callback which might add additional attributes to the login response.
+
+        Args:
+            login_result: The data to be sent to the client. Includes the user
+                ID and access token.
+        """
+        # Expire attributes before processing. Note that there shouldn't be any
+        # valid logins that still have extra attributes.
+        self._expire_sso_extra_attributes()
+
+        extra_attributes = self._extra_attributes.get(login_result["user_id"])
+        if extra_attributes:
+            login_result.update(
+                (("extra_attributes", extra_attributes.extra_attributes),)
```

Comment:
extra_attributes isn't really the best name. 😄 Better suggestions wanted!

Reply:
can we just dump it straight into the result?

Reply:
I actually had that originally but I was concerned that if someone added a field like `access_token` it would break the world. It is possible we can just document that though and ask people to not do that. :)

Reply:
I think I am inclined to do that tbh.

Reply:
I've made this change.

Reply:
> What if someone adds a field which we end up wanting to use in future?

I think that's too bad. 🤷